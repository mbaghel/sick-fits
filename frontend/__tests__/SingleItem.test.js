import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import wait from "waait";
import { MockedProvider } from "react-apollo/test-utils";
import { fakeItem } from "../lib/testUtils";
import SingleItem, { SINGLE_ITEM_QUERY } from "../components/SingleItem";

describe("<SingleItem />", () => {
  it("loads the correct data", async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        result: {
          data: {
            item: fakeItem()
          }
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    expect(wrapper.contains("Loading...")).toBe(true);

    await wait(0);
    wrapper.update();

    expect(toJSON(wrapper.find("img"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("h2"))).toMatchSnapshot();
    expect(toJSON(wrapper.find("p"))).toMatchSnapshot();
  });

  it("displays error if item not found", async () => {
    const mocks = [
      {
        request: { query: SINGLE_ITEM_QUERY, variables: { id: "123" } },
        result: { errors: [{ message: "Items Not Found!" }] }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <SingleItem id="123" />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    expect(
      toJSON(wrapper.find('[data-test="graphql-error"]'))
    ).toMatchSnapshot();
  });
});
