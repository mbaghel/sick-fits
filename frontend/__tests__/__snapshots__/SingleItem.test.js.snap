// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItem /> displays error if item not found 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Shoot!
  </strong>
  Items Not Found!
</p>
`;

exports[`<SingleItem /> loads the correct data 1`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;

exports[`<SingleItem /> loads the correct data 2`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;

exports[`<SingleItem /> loads the correct data 3`] = `
<p>
  dogs
</p>
`;
